rerun_py/rerun_sdk/rerun/time.py

```diff
@@ -1,12 +1,137 @@
+        Must be an integer.
+    timedelta:
+        Used for relative times, like `time_since_start`.
+        Must either be in seconds, a [`datetime.timedelta`][], or [`numpy.timedelta64`][].
+    datetime:
+        Used for absolute time indices, like `capture_time`.
+        Must either be in seconds since Unix epoch, a [`datetime.datetime`][], or [`numpy.datetime64`][].
+
+    """
+    if sum(x is not None for x in (sequence, timedelta, datetime)) != 1:
+        raise ValueError("Exactly one of `sequence`, `timedelta`, and `datetime` must be set (timeline='{timeline}')")
+
+    recording = recording.to_native() if recording is not None else None
+
+    if sequence is not None:
+        bindings.set_time_sequence(
+            timeline,
+            sequence,
+            recording=recording,
+        )
+    elif timedelta is not None:
+        if isinstance(timedelta, (int, float)):
+            nanos = int(1e9 * timedelta)  # Interpret as seconds and convert to nanos
+        else:
+            nanos = to_nanos(timedelta)
+        # TODO(#8635): call a function that is specific to time-deltas
+        bindings.set_time_nanos(
+            timeline,
+            nanos,
+            recording=recording,
+        )
+    elif datetime is not None:
+        if isinstance(datetime, (int, float)):
+            nanos = int(1e9 * datetime)  # Interpret as seconds and convert to nanos
+        else:
+            nanos = to_nanos_since_epoch(datetime)
+        # TODO(#8635): call a function that is specific to absolute times
+        bindings.set_time_nanos(
+            timeline,
+            nanos,
+            recording=recording,
+        )
+
+
+def to_nanos(timedelta_obj: int | float | timedelta | np.timedelta64) -> int:
+    if isinstance(timedelta_obj, (int, float)):
+        return int(timedelta_obj * 1e9)
+    elif isinstance(timedelta_obj, timedelta):
+        return int(timedelta_obj.total_seconds() * 1e9)
+    elif isinstance(timedelta_obj, np.timedelta64):
+        return timedelta_obj.astype("timedelta64[ns]").astype("int64")  # type: ignore[no-any-return]
+    else:
+        raise TypeError("timedelta_obj must be an int, float, timedelta, or numpy.timedelta64 object")
+
+
+def to_nanos_since_epoch(date_time: int | float | datetime | np.datetime64) -> int:
+    if isinstance(date_time, (int, float)):
+        return int(date_time * 1e9)
+    elif isinstance(date_time, datetime):
+        if date_time.tzinfo is None:
+            date_time = date_time.replace(tzinfo=timezone.utc)
+        else:
+            date_time = date_time.astimezone(timezone.utc)
+        epoch = datetime(1970, 1, 1, tzinfo=timezone.utc)
+        return int((date_time - epoch).total_seconds() * 1e9)
+    elif isinstance(date_time, np.datetime64):
+        return date_time.astype("int64")  # type: ignore[no-any-return]
+    else:
+        raise TypeError("date_time must be an int, float, datetime, or numpy.datetime64 object")
+
+
+# TODO(#8635): deprecate
```

Comment:
Any reason not to do that directly?


```suggestion
@deprecated(
    """Use `set_index()` instead.
    See: https://www.rerun.io/docs/reference/migration/migration-0-23?speculative-link for more details."""
)
```

Reply:
May be worth considering converting them all in this PR? Doesn't sound tooo bad.

Also, ideally, I'd rather have the test converted in this PR, and a new test added to cover for illegal calling patterns.

Reply:
(happy to help with that)

Reply:
Yeah, feel free to push some commits! 🙏  Let me know what you wanna work on though so we don't overlap

> May be worth considering converting them all in this PR?
Yeah, it's in the TODO of the PR description